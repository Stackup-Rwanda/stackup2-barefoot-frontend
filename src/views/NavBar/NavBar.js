import React from 'react';
import { makeStyles } from '@material-ui/core/styles';
import AppBar from '@material-ui/core/AppBar';
import Toolbar from '@material-ui/core/Toolbar';
import { Link } from 'react-router-dom';

const useStyles = makeStyles((theme) => ({
  root: {
    flexGrow: 1,
  },
  navlink: {
    color: theme.palette.primary.main,
  },
}));

const NavBar = () => {
  const classes = useStyles();

  return (
    <div className={classes.root}>
      <AppBar position="static" className="navbar">
        <Toolbar className="navbar-content">
          <div className="brand">
            <Link to="/" className="brand-link">
              <span className="brand-barefoot">Barefoot</span>
              <span className="brand-nomad">Nomad</span>
            </Link>
          </div>
          <div className="brand">
            <Link to="/" color="primary" className="navlink">Login</Link>
            <Link to="/dashboard/requests" color="textPrimary" className="navlink navlink-signup">Create account</Link>
          </div>
        </Toolbar>
      </AppBar>
    </div>
  );
};

export default NavBar;
