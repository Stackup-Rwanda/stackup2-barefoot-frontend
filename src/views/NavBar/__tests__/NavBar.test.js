/* eslint-disable no-undef */
import React from 'react';
import { BrowserRouter as Router } from 'react-router-dom';
import renderer from 'react-test-renderer';
import { cleanup } from '@testing-library/react';
import NavBar from '../NavBar';

describe('<NavBar/>', () => {
  afterEach(cleanup);
  it('Should match the NavBar component snapshot', () => {
    const tree = renderer.create(
      <Router>
        <NavBar />
      </Router>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
