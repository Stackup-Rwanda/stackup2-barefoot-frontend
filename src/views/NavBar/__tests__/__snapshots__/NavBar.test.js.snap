// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NavBar/> Should match the NavBar component snapshot 1`] = `
<div
  className="makeStyles-root-1"
>
  <header
    className="MuiPaper-root MuiAppBar-root MuiAppBar-positionStatic MuiAppBar-colorPrimary navbar MuiPaper-elevation4"
  >
    <div
      className="MuiToolbar-root MuiToolbar-regular navbar-content MuiToolbar-gutters"
    >
      <div
        className="brand"
      >
        <a
          className="brand-link"
          href="/"
          onClick={[Function]}
        >
          <span
            className="brand-barefoot"
          >
            Barefoot
          </span>
          <span
            className="brand-nomad"
          >
            Nomad
          </span>
        </a>
      </div>
      <div
        className="brand"
      >
        <a
          className="navlink"
          color="primary"
          href="/"
          onClick={[Function]}
        >
          Login
        </a>
        <a
          className="navlink navlink-signup"
          color="textPrimary"
          href="/dashboard/requests"
          onClick={[Function]}
        >
          Create account
        </a>
      </div>
    </div>
  </header>
</div>
`;
